.github/workflows/deploy-docs.yml

```diff
@@ -3,62 +3,173 @@ permissions:
+    pull_request_target:
+        types: [opened, synchronize, reopened, ready_for_review, labeled]
 
-env:
-  DOXYGEN_VERSION: 1.11.0
-  DOXYGEN_AWESOME_VERSION: 2.3.3
-        
 jobs:
-    build-and-deploy:
-        runs-on: ubuntu-latest
-        concurrency: ci-${{github. ref}}
-        steps: 
-            - name : Checkout
-              uses: actions/checkout@v4
-
-            - name: Install Clang
-              run: |
-                sudo apt-get update
-                sudo apt-get install -y clang
-            
-            - name: Install Doxygen 
-              shell: bash
-              run: |
-                wget https://github.com/doxygen/doxygen/releases/download/Release_${DOXYGEN_VERSION//./_}/doxygen-${DOXYGEN_VERSION}.linux.bin.tar.gz
-                tar xzvf doxygen-${DOXYGEN_VERSION}.linux.bin.tar.gz
-                cd doxygen-${DOXYGEN_VERSION}
-                sudo make install
-                cd ..
-                rm -rf doxygen-${DOXYGEN_VERSION}*  
-            
-            - name: Install graphviz
-              run: sudo apt-get install graphviz -y
-
-            - name: Clone Doxygen Awesome CSS Themes
-              working-directory: ./docs/doxygen
-              run: |
-                git clone --depth 1 -b v${{ env.DOXYGEN_AWESOME_VERSION }} https://github.com/jothepro/doxygen-awesome-css awesome_css
-
-            - name: Generate Doxygen Documentation
-              run: |
-                   cd docs/doxygen
-                   doxygen doxygen.cfg
-              shell: bash
-            
-            - name: Create .nojekyll 
-              run: |
-                    cd docs/doxygen/build
-                    touch html/.nojekyll
-              shell: bash
-
-            - name: Deploy to GitHub Pages
-              uses: JamesIves/github-pages-deploy-action@v4
-              with:
-                token: ${{ secrets.GITHUB_TOKEN }}
-                branch: gh-pages
-                folder: docs/doxygen/build/html
+  build-and-deploy:
+    runs-on: ubuntu-latest
+    env:
+      DOXYGEN_VERSION: 1.11.0
+      DOXYGEN_AWESOME_VERSION: 2.3.3
+      PR_PATH: pr-preview/${{ github.event.number }}
+      DOMAIN: ${{ github.actor }}.github.io
+      DEPLOYMENT: '<img src="https://github.com/user-attachments/assets/f94fada5-45ca-4271-9106-180728235ad2" alt="Rocket" width="25"/>'
+      CELEBRATION: '<img src="https://github.com/user-attachments/assets/86eb8470-c597-4f8b-a77d-a54036075271" alt="githubloading" width="25"/>'
+      FAILURE_ICON: '<img src="https://github.com/user-attachments/assets/af921ba9-953d-465e-b597-64ed0a2dc921" alt="failed" width="25"/>'
+      COMMENT_IDENTIFIER: '<!-- preview-comment -->'
+
+    concurrency: ci-${{ github.ref }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ github.event.pull_request.head.ref }}
+          repository: ${{ github.event.pull_request.head.repo.full_name }}
+
+      - name: Check PR Label
```

Comment:
This can be simpler. Take a look at other workflows: https://github.com/p4lang/p4c/blob/main/.github/workflows/ci-ubuntu-20-sanitizer-nightly.yml#L15

Reply:
I initially attempted to use the `pull_request` trigger, but it’s restricted by security rules for commenting on PR from forks. 

We need to use `pull_request_target` for commenting because it has the required access to the base repository. However, it doesn’t provide an easy way to fetch labels directly. To handle this, I used `actions/github-script` to retrieve labels through GitHub's REST API, which works for forked PRs.

Reply:
Hmm I see, this workflow is very complex it could be hard to maintain. 

Let's merge it and see how it behaves.